File: Medium/2026-02-23-1461-Check-If-a-String-Contains-All-Binary-Codes-of-Size-K/solution.js
var hasAllCodes = function(s, k) {
    const totalCodes = 1 << k; // 2^k
    const seen = new Set();

    for (let i = 0; i <= s.length - k; i++) {
        seen.add(s.substring(i, i + k));
        if (seen.size === totalCodes) {
            return true;
        }
    }

    return false;
};

File: Medium/2026-02-23-1461-Check-If-a-String-Contains-All-Binary-Codes-of-Size-K/solution.java
import java.util.HashSet;

public class Solution {
    public boolean hasAllCodes(String s, int k) {
        int totalCodes = 1 << k; // 2^k
        HashSet<String> seen = new HashSet<>();
        
        for (int i = 0; i <= s.length() - k; i++) {
            seen.add(s.substring(i, i + k));
            if (seen.size() == totalCodes) {
                return true;
            }
        }
        
        return false;
    }
}